xbmc/epg/GUIEPGGridContainer.cpp

```diff
@@ -1935,16 +1945,25 @@ void CGUIEPGGridContainer::GoToNow()
   int offset = ((currentDate - m_gridStart).GetSecondsTotal() / 60 - 30) / MINSPERBLOCK;
   ScrollToBlockOffset(offset);
 
-  for (int blockIndex = 0; blockIndex < m_blocksPerPage; blockIndex++)
+  if (m_channelCursor + m_channelOffset >= 0 &&
+      m_channelCursor + m_channelOffset < m_channels)
   {
-    const CFileItemPtr item = m_gridIndex[m_channelCursor + m_channelOffset][offset + blockIndex].item;
-    if (item)
+    for (int blockIndex = 0; blockIndex < m_blocksPerPage; blockIndex++)
     {
-      const CEpgInfoTagPtr tag = item->GetEPGInfoTag();
-      if (tag && tag->StartAsUTC() <= currentDate && tag->EndAsUTC() > currentDate)
-      {
-        SetBlock(blockIndex); // Select currently active epg element
+      if (offset + blockIndex < 0)
+        continue;
```

Comment:
This leads to selection of the epg event **after** the currently active epg event. Instead of this check + continue, I suggest to set offset to zero before entering the loop, if negative. Runtime tested, works like it should.